### ISSUE TITLE
docstring of BookBuilder should be more illustrative

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3120]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#3120|
|Type|enhancement|
|Created|2008-03-25 17:40:54Z|

I don't really know what arguments to pass to generate the book for Twisted.

It'd also be good to point out that the user may want to run ManBuilder before running BookBuilder.

<details><summary>Searchable metadata</summary>

```
trac-id__3120 3120
type__enhancement enhancement
reporter__radix radix
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1206466854000000 1206466854000000
changetime__1206973488000000 1206973488000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @exarkun|
|-|-|

How's this?

```
Index: twisted/python/_release.py
===================================================================
--- twisted/python/_release.py  (revision 23144)
+++ twisted/python/_release.py  (working copy)
@@ -326,6 +326,25 @@
 class BookBuilder(LoreBuilderMixin):
     """
     Generate the LaTeX and PDF documentation.
+
+    The book is built by assembling a number of LaTeX documents.  Only the
+    overall document which describes how to assemble the documents is stored
+    in LaTeX in the source.  The rest of the documentation is generated from
+    Lore input files.  These are primarily XHTML files (of the particular
+    Lore subset), but man pages are stored in GROFF format.  BookBuilder
+    expects all of its input to be Lore XHTML format, so L{ManBuilder}
+    should be invoked first if the man pages are to be included in the
+    result (this is determined by the book LaTeX definition file). 
+    Therefore, a sample usage of BookBuilder may look something like this:
+
+        man = ManBuilder()
+        man.build(FilePath("doc/core/man"))
+        book = BookBuilder()
+        book.build(
+            FilePath('doc/core/howto'),
+            [FilePath('doc/core/howto'), FilePath('doc/core/howto/tutorial'),
+             FilePath('doc/core/man'), FilePath('doc/core/specifications')],
+            FilePath('doc/core/howto/book.tex'), FilePath('/tmp/book.pdf'))
     """
     def run(self, command):
         """
@@ -346,7 +365,7 @@
 
     def buildTeX(self, howtoDir):
         """
-        Build LaTex files for lore input files in the given directory.
+        Build LaTeX files for lore input files in the given directory.
 
         Input files ending in .xhtml will be considered. Output will written as
         .tex files.
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @radix|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_5"></a>|@radix set owner to @therve|
|-|-|

That does look very helpful. Thanks, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @exarkun|
|-|-|

You're the one :).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [23151]) Apply BookBuilder docstring improvement patch

Author: exarkun
Reviewer: radix
Fixes [#3120](https://github.com/twisted/twisted/issues/3120)

Add an explanation of BookBuilder to that class's docstring, including an example of how it
might be used.  Also fix the capitalization of LaTeX in the docstring of a method of that class.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
